src/main/ingestion-queues/kafka-queue.ts

```diff
@@ -51,39 +44,40 @@ export class KafkaQueue implements Queue {
     private async eachEvent(event: PluginEvent): Promise<void> {
         const eachEventStartTimer = new Date()
 
-        const processingTimeout = timeoutGuard('Still running plugins on event. Timeout warning after 30 sec!', {
-            event: JSON.stringify(event),
-        })
-        const timer = new Date()
-        let processedEvent: PluginEvent
-        try {
-            processedEvent = await this.processEvent(event)
-        } catch (error) {
-            status.info('🔔', error)
-            Sentry.captureException(error)
-            throw error
-        } finally {
-            this.pluginsServer.statsd?.timing('kafka_queue.single_event', timer)
-            clearTimeout(processingTimeout)
+        let processedEvent: PluginEvent | null = event
+
+        if (processedEvent.event !== '$snapshot') {
```

Comment:
Absolute nit: since this is used in two places, the logic would be nicer to follow if the condition was a bool const `isSnapshot`

Reply:
agreed